test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java

```diff
@@ -904,7 +906,7 @@ private ClusterHealthStatus ensureGreenOrYellow(boolean green, TimeValue timeout
                 client().admin().cluster().preparePendingClusterTasks().get());
             fail("timed out waiting for " + color + " state");
         }
-        if (green) {
+        if (clusterHealthStatus == ClusterHealthStatus.GREEN) {
```

Comment:
nit: we can change this to greater than or equal relationship and check on both statuses.